plugins/mangadex/index.js

```diff
@@ -167,12 +167,19 @@ function selectChapter(id) {
 function nextPage() {
 	const page = parseInt(mango.storage('page'));
 	const atHome = JSON.parse(mango.storage('atHomeData'));
-	const filename = atHome.chapter.data[page]
-	mango.storage('page', (page + 1).toString());
+	const filename = atHome.chapter.data[page];
 	if (!filename) return JSON.stringify({});
 
+	//Get the number of 'numbers' of pages
+	const len = Math.ceil(Math.log(atHome.chapter.data.length + 1) / Math.LN10);
```

Comment:
Isn't it simpler to do `const len = atHome.chapter.data.length.toString().length`? The fact that `filename` is not undefined implies that `atHome.chapter.data.length` is positive.

Or if you prefer the arithmetic approach, you can simply do `Math.log10(atHome.chapter.data.length + 1)` without the extra step to convert from base 2.